lib/r10k/api/envmap_builder.rb

```diff
@@ -0,0 +1,52 @@
+require 'r10k/module/git'
+require 'r10k/module/svn'
+require 'r10k/module/forge'
+
+module R10K
+  module API
+    class EnvmapBuilder
+      def initialize
+        @modules = []
+      end
+
+      def build
+        @modules
+      end
+
+      # @param [String] forge
+      def set_forge(forge)
+        # No implementation for EnvMaps.
+      end
+
+      # @param [String] moduledir
+      def set_moduledir(moduledir)
+        # No implementation for EnvMaps.
+      end
+
+      # @param [String] name
+      # @param [Hash] args
+      def add_module(name, args)
+        name_parts = name.split(/[\-\/]/, 2)
+        mod_name = name_parts.size > 1 ? name_parts.last : name
+
+        if @modules.collect { |m| m[:name] }.include?(mod_name)
+          raise RuntimeError, "Puppetfile cannot declare the same module name twice: #{mod_name} was already declared"
```

Comment:
My inclination would be to make this an error going forward and since this code won't impact existing functionality I'm not sure we need to match the current functionality. This is the kind of thing that would make switching the CLI actions to use this code require a major version bump though. Maybe we should just add a comment or something documenting that this varies from existing behavior?